crypto/s2n_hmac.c

```diff
@@ -192,9 +192,9 @@ int s2n_hmac_reset(struct s2n_hmac_state *state)
     return 0;
 }
 
-int s2n_hmac_digest_verify(const void *a, uint32_t alen, const void *b, uint32_t blen)
+int s2n_hmac_digest_verify(const void *a, const void *b, uint32_t len)
 {
-    return 0 - (!s2n_constant_time_equals(a, b, alen) | !!(alen - blen));
+    return 0 - !s2n_constant_time_equals(a, b, len);
```

Comment:
Returning early when `alen` is different from `blen` is a possibility.  “constant-time” is only an idiomatic shortcut for “time independent of secrets”. `alen` and `blen` are not secrets, so it doesn't matter that the execution time depends on them. The contents pointed by `a` and `b` are secrets, but these cannot be observed in the time taken by any execution of `s2n_hmac_digest_verify`, regardless of whether the short path was taken or not.

However, the function is named `s2n_hmac_digest_verify`, and is intended to compare digests. It does not generally make sense to compare digests computed with differing algorithms, unless your name is Ange Albertini and you are working on your latest masterpiece. For everyone else, a digest computed with MD5 should only be compared to a digest computed with MD5, and digests computed with SHA3-256 should only be compared to digests computed with SHA3-256. Therefore there is only one length, the length of the digest algorithm being used on both sides.

Also, if `s2n_hmac_digest_verify` keeps its current prototype, tests are missing where it would be called directly or indirectly with differing `alen` and `blen`. Currently, despite the quality of the testsuite provided in the archive, the function is only ever invoked with the same argument for both.